src/orders/components/OrderPaymentSummaryCard/components/OrderPaymentStatusPill.tsx

```diff
@@ -0,0 +1,50 @@
+import { Pill } from "@dashboard/components/Pill";
+import { OrderChargeStatusEnum, OrderDetailsFragment } from "@dashboard/graphql";
+import { transformPaymentStatus } from "@dashboard/misc";
+import React from "react";
+import { useIntl } from "react-intl";
+
+type Order = Pick<OrderDetailsFragment, "paymentStatus" | "chargeStatus">;
+
+export interface OrderPaymentStatusPillProps {
+  order: Order | undefined;
+  className?: string;
+}
+
+export const OrderPaymentStatusPill = ({ order, className }: OrderPaymentStatusPillProps) => {
+  const intl = useIntl();
+
+  if (!order) {
+    return null;
+  }
+
+  const payment = transformPaymentStatus(order.paymentStatus, intl);
+
+  if (order.chargeStatus === OrderChargeStatusEnum.OVERCHARGED) {
+    return (
+      <Pill
+        key={payment.status}
+        label={intl.formatMessage({
```

Comment:
suggestion: You return `localized` from `transformPaymentStatus ` and you don't use it